include/i2c.h

```diff
@@ -175,6 +216,146 @@ static inline int _impl_i2c_transfer(struct device *dev,
 	return api->transfer(dev, msgs, num_msgs, addr);
 }
 
+#ifdef CONFIG_I2C_SLAVE
+/**
+ * @brief Returns if the current I2C Controller support Slave Mode
+ *
+ * This routine returns if the current I2C Controller support Slave Mode.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ *
+ * @retval false If not supported
+ * @retval true If supported
+ */
+__syscall bool i2c_slave_is_supported(struct device *dev);
+
+static inline bool _impl_i2c_slave_is_supported(struct device *dev)
```

Comment:
I don't think you need this.
user/dev should know what the controller he's working against should support. 

And, if he is really trying to mess with the wrong controller: i2c_slave_attach() should return -ENOTSUP (or -EBUSY if already used by another slave driver) and that's it.

Reply:
You are right, this call is quite useless indeed